### DISCUSSION TITLE
next.config.js 301 Redirect Not Working

### DISCUSSION THREAD

**Discussion by tcagkansokmen** [vercel/next.js#29478] [Help]
I want to redirect search?q=blabla to search/blabla.
I tried so many thing but nothing worked. My last code is below.

CAn anyone help? There's nothing on internet about it, and not enough examples on nextjs documentation.

```
{
        source: '/search',
        has: [
          {
            type: 'query',
            key: 'q',
            value: '(?<paramName>.*)',
          },
        ],
        permanent: false,
        destination: '/search/:paramName',
      },
```

**Comment by joggienl** [comment]
I posted this question yesterday, no anwsers yet but it might be related: https://github.com/vercel/next.js/discussions/29475

**Comment by wwwdepot** [comment]
Redirects will bring query from source over to new destination, so `/search?q=123` becomes `/search/123?q=123`. 

If you want clean redirect (no querystring), you should create SSR page `/search/index.js`

```js
export async function getServerSideProps(context) {
  return {
    redirect: {
      destination: "/search/" + context.query.q,
      permanent: false,
    },
  };
}
```
